File: app/src/main/java/app/axon/test/ui/base/BaseContract.java
package app.axon.test.ui.base;

import androidx.annotation.StringRes;

public class BaseContract {


    public interface Presenter<T> {
        void attach(T view);

    }

    public interface View {

        void showMessage(String message);

        void showMessage(@StringRes int text);

        void showProgress();

        void hideProgress();

        void showSnackbar();
    }

}

File: app/src/main/java/app/axon/test/ui/base/BaseContract.kt
package app.axon.test.ui.base

open class BaseContract {

    interface Presenter<T : View> {
        fun attach(view: T)

        fun detach()
    }

    interface View {

        fun showMessage(message: String)

        fun showMessage(message: Int)

        fun showProgress(){}

        fun hideProgress(){}

        fun showNoInternetConnection(){}

        fun showError() {}

    }

}